wire-ios-data-model/Tests/Source/Model/Messages/ZMClientMessagesTests+Reaction.swift

```diff
@@ -139,6 +182,74 @@ extension ZMClientMessageTests_Reaction {
         XCTAssertTrue(message.cachedCategory.contains(.text))
-        XCTAssertFalse(message.cachedCategory.contains(.liked))
+        XCTAssertFalse(message.cachedCategory.contains(.reacted))
     }
+
+    func testThatANessageWithReactionsWhenReceivingUpdateEventWithNoReactionsRemovesAllReactions() {
+        // GIVEN
+        let message = insertMessage()
+        ZMMessage.addReaction("❤️", to: message)
+        ZMMessage.addReaction("😍", to: message)
+        uiMOC.saveOrRollback()
+
+        let event = updateEventForRemovingReaction(to: message)
+
+        // WHEN
+        performPretendingUiMocIsSyncMoc {
+            ZMClientMessage.createOrUpdate(from: event, in: self.uiMOC, prefetchResult: nil)
+        }
+        XCTAssertTrue(uiMOC.saveOrRollback())
+        XCTAssertTrue(waitForAllGroupsToBeEmpty(withTimeout: 0.5))
+
+        // THEN
+        XCTAssertEqual(message.reactions.count, 0)
+        XCTAssertEqual(message.usersReaction.count, 0)
+    }
+
+    func testThatAMessageWithReactionsWhenReceivingUpdateEventWithNewReactionsUpdatesReactions() {
+        // GIVEN
+        let message = insertMessage()
+        ZMMessage.addReaction("❤️", to: message)
+        ZMMessage.addReaction("😍", to: message)
+        uiMOC.saveOrRollback()
+
+        let genericMessage = GenericMessage(content: WireProtos.Reaction.createReaction(emojis: ["🥰", "😃", "❤️", "😍"], messageID: message.nonce!))
+
+        let event = createUpdateEvent(UUID(), conversationID: conversation.remoteIdentifier!, genericMessage: genericMessage, senderID: message.sender!.remoteIdentifier!)
+
+        // WHEN
+        performPretendingUiMocIsSyncMoc {
+            ZMClientMessage.createOrUpdate(from: event, in: self.uiMOC, prefetchResult: nil)
+        }
+        XCTAssertTrue(uiMOC.saveOrRollback())
+        XCTAssertTrue(waitForAllGroupsToBeEmpty(withTimeout: 0.5))
+
+        // THEN
+        XCTAssertEqual(message.reactions.count, 4)
+        XCTAssertEqual(message.usersReaction.count, 4)
+    }
+
+    func testThatAMessageWithAReactionWhenReceivingUpdateEventWithANewReactionItOnlyContainsTheNewReaction() {
+        // GIVEN
+        let message = insertMessage()
+        ZMMessage.addReaction("❤️", to: message)
+        uiMOC.saveOrRollback()
+
+        let genericMessage = GenericMessage(content: WireProtos.Reaction.createReaction(emojis: ["🥰"], messageID: message.nonce!))
+
+        let event = createUpdateEvent(UUID(), conversationID: conversation.remoteIdentifier!, genericMessage: genericMessage, senderID: message.sender!.remoteIdentifier!)
+
+        // WHEN
+        performPretendingUiMocIsSyncMoc {
+            ZMClientMessage.createOrUpdate(from: event, in: self.uiMOC, prefetchResult: nil)
+        }
+        XCTAssertTrue(uiMOC.saveOrRollback())
+        XCTAssertTrue(waitForAllGroupsToBeEmpty(withTimeout: 0.5))
+
+        // THEN
+        XCTAssert(message.usersReaction.contains(where: { (key: String, _) in
+            key == "🥰"
+        }))
```

Comment:
You should be asserting who reacted too.

Reply:
Done!